src/Aspire.Hosting/Orchestrator/ApplicationOrchestrator.cs

```diff
@@ -117,6 +117,11 @@ private async Task OnEndpointsAllocated(OnEndpointsAllocatedContext context)
         {
             await lifecycleHook.AfterEndpointsAllocatedAsync(_model, context.CancellationToken).ConfigureAwait(false);
         }
+
+        foreach (var resource in _model.Resources.OfType<IResourceWithoutLifetime>())
```

Comment:
Undo this change and put this back where it was. The layering was better.

Reply:
Think I have a better solution now, although it does rely on treating `ParameterResource` and `ConnectionStringExpression` specifically rather than generically handle all `IResourceWithoutLifetimes` that are also `IManifestExpressionProvider`.  Think it's good, will do some more validation tomorrow.

This was supposed to be a one-liner when I first started 😅.